src/main/java/com/squareup/protoparser/ProtoParser.java

```diff
@@ -842,25 +842,58 @@ private String tryAppendTrailingDocumentation(String documentation) {
       }
     }
 
-    if (pos == data.length || data[pos] != '/') {
-      throw unexpected("expected '/'");
+    if (pos == data.length || (data[pos] != '/' && data[pos] != '*')) {
+      pos--; // Backtrack to start of comment.
+      throw unexpected("expected '//' or '/*'");
     }
+    boolean isStar = data[pos] == '*';
     pos++;
+
     if (pos < data.length && data[pos] == ' ') {
       pos++; // Skip a single leading space, if present.
     }
 
-    // Consume comment until newline.
     int start = pos;
-    while (pos < data.length) {
-      char c = data[pos++];
-      if (c == '\n') {
-        newline();
-        break;
+    int end;
+
+    if (isStar) {
+      // Consume multi-line comment until it closes on the same line.
+      while (true) {
+        if (pos == data.length || data[pos] == '\n') {
+          throw unexpected("missing '*/' for trailing '/*' comment");
+        }
+        if (data[pos] == '*' && pos < data.length - 1 && data[pos + 1] == '/') {
+          end = pos - 1;
+          pos += 2;
+          break;
+        }
+        pos++;
+      }
+    } else {
+      // Consume comment until newline.
+      while (true) {
+        if (pos == data.length) {
+          end = pos - 1;
+          break;
+        }
+        char c = data[pos++];
+        if (c == '\n') {
+          newline();
+          end = pos - 2;
```

Comment:
The `- 2` is because `newLine()` expects to be on the character after the newline instead of on the newline–a potentially confusing behavior itself.